src/main/java/com/trustev/web/ApiClient.java

```diff
@@ -22,727 +21,1387 @@
+     * @param emailId The id of the Email you want to update
+     * @return the updated Email object
+     * @throws TrustevApiException A Custom Trustev Api Exception
+     */
+    public static Email updateEmail(String caseId, Email email, String emailId) throws TrustevApiException {
+        String url = "/Case/{id}/Customer/Email/{id2}".replace("{id}", caseId).replace("{id2}", emailId);
+        ;
+        Email response = (Email) PerformHttpCall(url, HttpMethod.PUT, Email.class, email, true);
+        return response;
+    }
+
+    /**
+     * Update a specific Email on a Case which already contains a Email
+     *
+     * @param caseId   The Case Id of a Case which you have already posted
+     * @param email    The Email you want to update the existing Email to
+     * @param emailId  The id of the Email you want to update
+     * @param username The merchant site username that the request is being made by
+     * @return the updated Email object
+     * @throws TrustevApiException A Custom Trustev Api Exception
+     */
+    public static Email updateEmail(String caseId, Email email, String emailId, String username) throws TrustevApiException {
+        String url = "/Case/{id}/Customer/Email/{id2}".replace("{id}", caseId).replace("{id2}", emailId);
+        ;
+        Email response = (Email) PerformHttpCall(url, HttpMethod.PUT, Email.class, email, true, username);
+        return response;
+    }
+
+    /**
+     * Get a specific Email from a Case
+     *
+     * @param caseId  The Case Id of a Case with the Customer which you have already posted
+     * @param emailId The Id of the Email you want to get
+     * @return the Email object which id matches caseId
+     * @throws TrustevApiException A Custom Trustev Api Exception
+     */
+    public static Email getEmail(String caseId, String emailId) throws TrustevApiException {
+        String url = "/Case/{id}/Customer/Email/{id2}".replace("{id}", caseId).replace("{id2}", emailId);
+        ;
+        Email response = (Email) PerformHttpCall(url, HttpMethod.GET, Email.class, null, true);
+        return response;
+    }
+
+    /**
+     * Get a specific Email from a Case
+     *
+     * @param caseId   The Case Id of a Case with the Customer which you have already posted
+     * @param emailId  The Id of the Email you want to get
+     * @param username The merchant site username that the request is being made by
+     * @return the Email object which id matches caseId
+     * @throws TrustevApiException A Custom Trustev Api Exception
+     */
+    public static Email getEmail(String caseId, String emailId, String username) throws TrustevApiException {
+        String url = "/Case/{id}/Customer/Email/{id2}".replace("{id}", caseId).replace("{id2}", emailId);
+        ;
+        Email response = (Email) PerformHttpCall(url, HttpMethod.GET, Email.class, null, true, username);
+        return response;
+    }
+
+    /**
+     * Get all the Emails from
```

Comment:
What if in this method we get the unique username and call return PerformHttpCall(uriPath,httpMethod,responseType,entity,isAuthenticationNeeded,userName);  ?? To avoid repeated code.